homeassistant/components/ecobee/climate.py

```diff
@@ -678,15 +679,32 @@ def resume_program(self, resume_all):
 
     def hold_preference(self):
         """Return user preference setting for hold time."""
-        # Values returned from thermostat are 'useEndTime4hour',
-        # 'useEndTime2hour', 'nextTransition', 'indefinite', 'askMe'
-        default = self.thermostat["settings"]["holdAction"]
-        if default == "nextTransition":
-            return default
-        # add further conditions if other hold durations should be
-        # supported; note that this should not include 'indefinite'
-        # as an indefinite away hold is interpreted as away_mode
-        return "nextTransition"
+        # Values returned from thermostat are:
+        #   "useEndTime2hour", "useEndTime4hour"
+        #   "nextPeriod", "askMe"
+        #   "indefinite"
+        device_preference = self.thermostat["settings"]["holdAction"]
+        # Currently supported pyecobee holdTypes:
+        #   dateTime, nextTransition, indefinite, holdHours
+        hold_pref_map = {
```

Comment:
Since this is a constant, please move it to the top and upper case it like `ECOBEE_HVAC_ACTION_TO_HASS`

Reply:
I'll make the change, but believe that it will make the code less clear.  The current implementation has the map and usage together.  The suggested change separates them.  Additionally, the map is not needed outside this function.

Reply:
Let's ask for a second opinion on this.

Reply:
I'm with @jamincollins here. It would make the code less clear and since it doesn't get used anywhere else it shold be fine

Reply:
I'm going to the approval from the code owner and lack of objection as acceptance they they approve of the approach and remove the request for second opinion.